code/datums/ai/monkey/monkey_behaviours.dm

```diff
@@ -0,0 +1,262 @@
+			monkey_attack(controller, target, delta_time)
+		else
+			living_pawn.a_intent = INTENT_HARM
+			monkey_attack(controller, target, delta_time)
+
+
+/datum/ai_behavior/monkey_attack_mob/finish_action(datum/ai_controller/controller, succeeded)
+	. = ..()
+	controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET] = null
+
+/// attack using a held weapon otherwise bite the enemy, then if we are angry there is a chance we might calm down a little
+/datum/ai_behavior/monkey_attack_mob/proc/monkey_attack(datum/ai_controller/controller, mob/living/target, delta_time)
+
+	var/mob/living/living_pawn = controller.pawn
+
+	if(living_pawn.next_move > world.time)
+		return
+
+	living_pawn.changeNext_move(CLICK_CD_MELEE) //We play fair
+
+	var/obj/item/weapon = locate(/obj/item) in living_pawn.held_items
+
+	living_pawn.face_atom(target)
+
+	// attack with weapon if we have one
+	if(weapon)
+		weapon.melee_attack_chain(living_pawn, target)
+	else
+		target.attack_paw(living_pawn)
+
+	// no de-aggro
+	if(controller.blackboard[BB_MONKEY_AGRESSIVE])
+		return
+
+	if(DT_PROB(MONKEY_HATRED_REDUCTION_PROB, delta_time))
+		controller.blackboard[BB_MONKEY_ENEMIES][target]--
+
+	// if we are not angry at our target, go back to idle
+	if(controller.blackboard[BB_MONKEY_ENEMIES][target] <= 0)
+		var/list/enemies = controller.blackboard[BB_MONKEY_ENEMIES]
+		enemies.Remove(target)
+		if(controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET] == target)
+			finish_action(controller, TRUE)
+
+/datum/ai_behavior/disposal_mob
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_MOVE_AND_PERFORM //performs to increase frustration
+
+/datum/ai_behavior/disposal_mob/finish_action(datum/ai_controller/controller, succeeded)
+	. = ..()
+	controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET] = null //Reset attack target
+	controller.blackboard[BB_MONKEY_DISPOSING] = FALSE //No longer disposing
+	controller.blackboard[BB_MONKEY_TARGET_DISPOSAL] = null //No target disposal
+
+/datum/ai_behavior/disposal_mob/perform(delta_time, datum/ai_controller/controller)
+	. = ..()
+
+	if(controller.blackboard[BB_MONKEY_DISPOSING]) //We are disposing, don't do ANYTHING!!!!
+		return
+
+	var/mob/living/target = controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET]
+	var/mob/living/living_pawn = controller.pawn
+
+	controller.current_movement_target = target
+
+	if(target.pulledby != living_pawn && !HAS_AI_CONTROLLER_TYPE(target.pulledby, /datum/ai_controller/monkey)) //Dont steal from my fellow monkeys.
+		if(living_pawn.Adjacent(target) && isturf(target.loc))
+			living_pawn.a_intent = INTENT_GRAB
+			target.grabbedby(living_pawn)
+		return //Do the rest next turn
+
+	var/obj/machinery/disposal/disposal = controller.blackboard[BB_MONKEY_TARGET_DISPOSAL]
+	controller.current_movement_target = disposal
+
+	if(living_pawn.Adjacent(disposal))
+		INVOKE_ASYNC(src, .proc/try_disposal_mob, controller) //put him in!
+	else //This means we might be getting pissed!
+		return
```

Comment:
```suggestion
	if(living_pawn.Adjacent(disposal))
		INVOKE_ASYNC(src, .proc/try_disposal_mob, controller) //put him in!
		return	
```

Reply:
Can't I just ignore the else return block completely since it's the end of the proc?

Reply:
actually yes